core/src/Kokkos_Array.hpp

```diff
@@ -133,6 +133,17 @@ struct Array {
   KOKKOS_INLINE_FUNCTION constexpr const_pointer data() const {
     return &m_internal_implementation_private_member_data[0];
   }
+
+ private:
+  template <class U = T>
```

Comment:
It seems the template argument `U` is not used (?). I may be missing something.

Reply:
You're right I should have used in the SFINAE condition below instead of `T`

Reply:
Not sure why the compiler didn't yell at me